library/src/com/google/maps/android/geojson/GeoJsonLayer.java

```diff
@@ -71,6 +74,69 @@ public GeoJsonLayer(GoogleMap map, int resourceId, Context context)
     }
 
     /**
+     * Sets a single click listener for the entire GoogleMap object, that will be called
+     * with the corresponding GeoJsonFeature object when an object on the map (Polygon,
+     * Marker, Polyline) is clicked.
+     *
+     * Note that if multiple GeoJsonLayer objects are bound to a GoogleMap object, calling
+     * setOnFeatureClickListener on one will override the listener defined on the other. In
+     * that case, you must define each of the GoogleMap click listeners manually
+     * (OnPolygonClickListener, OnMarkerClickListener, OnPolylineClickListener), and then
+     * use the GeoJsonLayer.getFeature(mapObject) method on each GeoJsonLayer instance to
+     * determine if the given mapObject belongs to the layer.
+     *
+     * @param listener GeoJsonOnFeatureClickListener object providing the onFeatureClick
+     *                 method to call.
+     */
+    public void setOnFeatureClickListener(final GeoJsonOnFeatureClickListener listener) {
+
+        GoogleMap map = getMap();
+
+        map.setOnPolygonClickListener(new GoogleMap.OnPolygonClickListener() {
+            @Override
+            public void onPolygonClick(Polygon polygon) {
+                listener.onFeatureClick(getFeature(polygon));
+            }
+        });
+
+        map.setOnMarkerClickListener(new GoogleMap.OnMarkerClickListener() {
+            @Override
+            public boolean onMarkerClick(Marker marker) {
+                listener.onFeatureClick(getFeature(marker));
+                return false;
+            }
+        });
+
+        map.setOnPolylineClickListener(new GoogleMap.OnPolylineClickListener() {
+            @Override
+            public void onPolylineClick(Polyline polyline) {
+                listener.onFeatureClick(getFeature(polyline));
+            }
+        });
+
+    }
+
+    /**
+     * Callback interface for when a GeoJsonLayer's map object is clicked.
+     */
+    public interface GeoJsonOnFeatureClickListener {
+        void onFeatureClick(GeoJsonFeature feature);
+    }
+
+    /**
+     * Retrieves a corresponding GeoJsonFeature instance for the given map object
+     * (Polygon, Marker, Polyline). Made available so that maps with multiple layers
+     * can implement their own click listeners and determine which layer the
```

Comment:
I think this is useful beyond just click handling, so you should drop or reword to something like "Useful for maps with multiple..."